File: golang/04_control_flow.go
// 04_control_flow.go
// 主题：流程控制（if、for、switch）
// 本示例演示 Go 语言中的 if、for、switch 语句。
// 注意：如需运行本文件，请确保同目录下没有其他 main 包的文件。

package main

import "fmt"

func main() {
    // if 语句
    x := 7
    if x < 5 {
        fmt.Println("x 小于 5")
    } else {
        fmt.Println("x 大于等于 5")
    }

    // for 循环
    for i := 0; i < 3; i++ {
        fmt.Println("for 循环 i:", i)
    }

    // switch 语句
    day := 3
    switch day {
    case 1:
        fmt.Println("星期一")
    case 2:
        fmt.Println("星期二")
    case 3:
        fmt.Println("星期三")
    default:
        fmt.Println("其他")
    }
} 

File: rust/04_control_flow.rs
// 04_control_flow.rs
// 主题：流程控制
// 本示例演示 Rust 中的 if、loop、while 和 for 循环。

fn main() {
    // if 表达式
    let number = 7;
    if number < 5 {
        println!("小于 5");
    } else {
        println!("大于等于 5");
    }

    // loop 循环
    let mut count = 0;
    loop {
        count += 1;
        if count == 3 {
            println!("loop 计数到 3，跳出循环");
            break;
        }
    }

    // while 循环
    let mut n = 0;
    while n < 3 {
        println!("while 循环: n = {}", n);
        n += 1;
    }

    // for 循环
    for i in 0..3 {
        println!("for 循环: i = {}", i);
    }
} 